geopandas/base.py

```diff
@@ -54,6 +54,8 @@ def _delegate_binary_method(op, this, other, align, *args, **kwargs):
         other = GeometryArray(other.values)
     elif isinstance(other, BaseGeometry):
         a_this = GeometryArray(this.values)
+    elif isinstance(other, pd.Series) and isinstance(other.dtype, GeometryDtype):
+        a_this = GeometryArray(this.values)
```

Comment:
Can you combine this check with `if isinstance(other, GeoPandasBase)` above? - I think we should use the same align behaviour regardless of other being series/geoseries

Reply:
`AttributeError: 'Series' object has no attribute 'geometry'` if we pass in a regular pandas Series, but I can add some logic that constructs a GeoSeries out of `other` so that the same alignment logic can be used.